assets/model_monitoring/components/src/shared_utilities/io_utils.py

```diff
@@ -14,45 +24,60 @@ def init_spark():
     return spark
 
 
-def _is_not_found_exception(error: Exception):
-    return (
-        isinstance(error, IndexError)
-        or "The requested stream was not found" in error.args[0]
-        or "Not able to find MLTable file" in error.args[0]
-    )
+def _is_not_found_exception(input_name, error: Exception):
+    if isinstance(error, IndexError) or ("Partition 0 is out of bounds." in error.args[0]):
+        return True, (f"No data is found for input '{input_name}' in the specified window, "
+                      "most likely there is no request in the specified window.")
+    elif "The requested stream was not found" in error.args[0]:
```

Comment:
Do we know which data is missing?

Reply:
added mltable path to the error msg so user can check the MLTable content to get all the details.